extensions/amp-date-picker/1.0/base-element.js

```diff
@@ -0,0 +1,73 @@
+import {parseBooleanAttribute} from '#core/dom';
+
+import {PreactBaseElement} from '#preact/base-element';
+
+import {BentoDatePicker} from './component';
+import {CSS as COMPONENT_CSS} from './component.jss';
+import {parseDate, parseDateList, parseLocale} from './parsers';
+
+export class BaseElement extends PreactBaseElement {}
+
+/** @override */
+BaseElement['Component'] = BentoDatePicker;
+
+/** @override */
+BaseElement['props'] = {
+  'children': {passthrough: true},
+  'mode': {attr: 'mode'},
+  'type': {attr: 'type'},
+  'inputSelector': {attr: 'input-selector'},
+  'startInputSelector': {attr: 'start-input-selector'},
+  'end-input-selector': {attr: 'end-input-selector'},
+  'min': {attr: 'min', parseAttr: parseDate},
+  'max': {attr: 'max', parseAttr: parseDate},
+  'monthFormat': {attr: 'month-format'},
+  'format': {attr: 'format'},
+  'weekDayFormat': {attr: 'week-day-format'},
+  'locale': {attr: 'locale', parseAttr: parseLocale},
+  'maximumNights': {attr: 'maximum-nights', parseAttr: parseInt},
+  'minimumNights': {attr: 'minimum-nights', parseAttr: parseInt},
+  'numberOfMonths': {attr: 'number-of-months', parseAttr: parseInt},
+  // Not implemented
+  'firstDayOfWeek': {attr: 'first-day-of-week', parseAttr: parseInt},
```

Comment:
This feature is currently in-progress in `react-day-picker` beta. https://github.com/gpbl/react-day-picker/issues/1304